src/lua/zencode_pvss.lua

```diff
@@ -0,0 +1,227 @@
+        'pvss public key is not valid (not in the subgroup)'
+    )
+    return point
+end
+
+local function import_public_shares_f(obj)
+    local res = {}
+    res.public_keys = ZEN.get(obj, 'public_keys', pvss_public_key_f)
+    res.proof = ZEN.get(obj, 'proof', BIG.new, O.from_base64)
+    res.commitments = ZEN.get(obj, 'commitments', O.zcash_topoint, O.from_base64)
+    res.encrypted_shares = ZEN.get(obj, 'encrypted_shares', O.zcash_topoint, O.from_base64)
+    return res
+end
+
+local function export_public_shares_f(obj)
+    local res = {["public_keys"] = {}, ["proof"] = {}, ["encrypted_shares"] = {}, ["commitments"] = {}}
+    for i = 1, #obj.commitments do
+        res.commitments[i] = obj.commitments[i]:zcash_export()
+    end
+    local n = #obj.public_keys
+    for i=1,n do
+        res.public_keys[i] = obj.public_keys[i]:zcash_export()
+        res.proof[i] = obj.proof[i]:octet():base64()
+        res.encrypted_shares[i] = obj.encrypted_shares[i]:zcash_export()
+    end
+    res.proof[n+1] = obj.proof[n+1]:octet():base64()
+    return res
+end
+
+local function import_secret_share_f(obj)
+    local res = {}
+    res.index = ZEN.get(obj, 'index', INT.from_decimal, tostring)
+    res.proof = ZEN.get(obj, 'proof', BIG.new, O.from_base64)
+    res.dec_share = ZEN.get(obj, 'dec_share', O.zcash_topoint, O.from_base64)
+    res.enc_share = ZEN.get(obj, 'enc_share', O.zcash_topoint, O.from_base64)
+    res.pub_key = ZEN.get(obj, 'pub_key', pvss_public_key_f)
+    return res
+end
+
+local function export_secret_share_f(obj)
+    local res = {["proof"]={}}
+    res.index = obj.index:decimal()
+    res.enc_share = obj.enc_share:zcash_export()
+    res.dec_share = obj.dec_share:zcash_export()
+    res.pub_key = obj.pub_key:zcash_export()
+    for i=1,#obj.proof do
+        res.proof[i] = obj.proof[i]:octet():base64()
+    end
+    return res
+end
+
+local function import_verified_shares_f(obj)
+    local res = {}
+    res.valid_indexes = ZEN.get(obj, 'valid_indexes', INT.from_decimal, tostring)
+    res.valid_shares = ZEN.get(obj, 'valid_shares', O.zcash_topoint, O.from_base64)
+    return res
+end
+
+local function export_verified_shares_f(obj)
+    local res = {["valid_indexes"] = {}, ["valid_shares"] = {}}
+    for i = 1, #obj.valid_indexes do
+        res.valid_shares[i] = obj.valid_shares[i]:zcash_export()
+        res.valid_indexes[i] = obj.valid_indexes[i]:decimal()
+    end
+    return res
+end
+
+local function check_inputs(pks, commitments, encrypted_shares, proof, t, n)
+    ZEN.assert(t <= n, "Quorum is bigger than total")
+
+    ZEN.assert(type(pks) == 'table', "'pvss participant pks' is not a table")
+    ZEN.assert(type(commitments) == 'table', "'pvss commitments' is not a table")
+    ZEN.assert(type(encrypted_shares) == 'table', "'pvss encrypted shares' is not a table")
+    ZEN.assert(type(proof) == 'table', "'pvss proof' is not a table")
+
+    ZEN.assert(#pks == n, "'pvss participant pks' is of wrong length")
```

Comment:
These checks seem fairly important for input validation, therefore the overloaded operator `#` should not be used to check the length of tables. Use of `pairs` in a iteration is more trustworthy, see https://stackoverflow.com/questions/2705793/how-to-get-number-of-entries-in-a-lua-table

For example:
```
function tablelength(T)
  local count = 0
  for _ in pairs(T) do count = count + 1 end
  return count
end
```